crates/ra_syntax/src/syntax_error.rs

```diff
@@ -79,6 +83,10 @@ impl fmt::Display for SyntaxError {
 pub enum SyntaxErrorKind {
     ParseError(ParseError),
     EscapeError(EscapeError),
+    TokenizeError(TokenizeError),
+    // FIXME: the obvious pattern of this enum dictates that the following enum variants
+    // should be wrapped into something like `SemmanticError(SemmanticError)`
+    // or `ValidateError(ValidateError)` or `SemmanticValidateError(...)`
```

Comment:
Lets leave it at the tokenizer changes, at least for this PR. Then you can do this as a separate PR if it seems useful.

Reply:
Yes, I described the summary and future todos in the comment [under the original issue](https://github.com/rust-analyzer/rust-analyzer/issues/223#issuecomment-578559627). I'll see what should be done in next PR.